mautic/mautic
# Improved Searching by email - assumes email is lower. For large lead …

…databases, the migration is likely to take a long time to run to set all emails to lower(email)

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Implements  #5917 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description: Sets all emails to lower. When searching by email, doesn't use lower(email) anymore. Significant performance increase with large DB.

- With the full scan, the DB is having to read each record, run lower(email), then compare. Relatively intensive, especially with 100k + records.
- The index doesn't require any processing by the DB, so naturally faster.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Have large lead DB (1million + rows)
2.  Search by email. Can take significant time to find.

#### Steps to test this PR:
1.  Create db with test data of 1 million leads
2.  Search via email in contacts. 
3. Log/monitor queries. Use a programme such as PHPMyAdmin under performance monitor. Before the fix,  should see the query taking seconds to run. After the fix, should be faster (and may not see in query monitor). Ideally, turn on long query logging with a run time of 1 second or longer to view.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1.
2.
